awssdk/src/main/scala/implicits.scala

```diff
@@ -29,4 +31,19 @@ private[meteor] object implicits extends syntax {
       m3
     }
   }
+
+  implicit class RichDeleteItemRequestBuilder(val underlying: DeleteItemRequest.Builder) {
```

Comment:
> I would prefer to move this implicit class to a private method in DeleteOps as it is only being used there. 

I intend for this implicit to be subsequently used in other parts of the codebase too, to avoid code duplication.

> Scala 3 removed implicit so I would prefer to avoid it when possible. 

Scala 3 has an extension method mechanism, and the translation from implicit classes to that is fairly mechanical.

> Additionally, the style of the project is fairly functional, hence, would you mind changing the if-without-else into a full if-else please?

The underlying builder classes _are_ mutable. Pretending they're immutable can be misleading to maintainers, and can introduce subtle bugs.

Reply:
>I intend for this implicit to be subsequently used in other parts of the codebase too, to avoid code duplication.
that is fine but `DeleteItemRequest` is going to be used in `DeleteOps.scala` only but not any where else though.

>Scala 3 has an extension method mechanism, and the translation from implicit classes to that is fairly mechanical.
>The underlying builder classes are mutable. Pretending they're immutable can be misleading to maintainers, and can introduce subtle bugs.
those are good points. I'm happy to keep the implimentation as is but do you think `DeleteOps.scala` is a better place for it?